challenge.go

```diff
@@ -2,43 +2,107 @@ package srp
 
 import (
 	"bytes"
+	"context"
+	"crypto/hmac"
 	"crypto/sha256"
 	"encoding/base64"
 	"encoding/binary"
 	"errors"
 
+	"golang.org/x/crypto/argon2"
 	"golang.org/x/crypto/curve25519"
 )
 
-const ECDLPEphemeralSize = 24
+const ecdlpPRFKeySize = 32
 
-func ECDLPChallenge(b64Challenge string) (solution int64, err error) {
-	if len(b64Challenge) != 76 { // 56 bytes in base64
-		return 0, errors.New("srp:invalid ECDLP challenge length")
+func ECDLPChallenge(b64Challenge string, ctx context.Context) (b64Solution string, err error) {
```

Comment:
I highly doubt this is going to work with go-mobile. Can we use a simpler solution like an int with seconds?